mapzen-android-sdk/src/main/java/com/mapzen/android/routing/RouterInitializer.java

```diff
@@ -1,43 +1,37 @@
 package com.mapzen.android.routing;
 
-import com.mapzen.android.core.MapzenManager;
-
 import android.content.Context;
 
 import java.util.Map;
 
 /**
- * Handles setting the api key and a request handler for given {@link MapzenRouter} objects.
+ * Handles setting the request handler for given {@link MapzenRouter} objects.
  */
 public class RouterInitializer {
-  final MapzenRouterHttpHandler requestHandler = new MapzenRouterHttpHandler() {
-    @Override public Map<String, String> queryParamsForRequest() {
-      return null;
-    }
 
-    @Override public Map<String, String> headersForRequest() {
-      return null;
-    }
-  };
+  final MapzenRouterHttpHandler requestHandler;
 
   /**
-   * Initialize the {@link MapzenRouter}'s api key from mapzen.xml and set it's
-   * {@link MapzenRouterHttpHandler}.
-   * @param router
+   * Constructor.
    * @param context
    */
-  public void initRouter(MapzenRouter router, Context context) {
-    initRouter(router, MapzenManager.instance(context).getApiKey());
+  public RouterInitializer(Context context) {
+    requestHandler = new MapzenRouterHttpHandler(context) {
+      @Override public Map<String, String> queryParamsForRequest() {
+        return null;
+      }
+
+      @Override public Map<String, String> headersForRequest() {
+        return null;
+      }
+    };
   }
 
   /**
-   * Initialize the {@link MapzenRouter}'s api key in code and set it's
-   * {@link MapzenRouterHttpHandler}.
+   * Set the {@link MapzenRouter}'s {@link MapzenRouterHttpHandler}.
    * @param router
-   * @param apiKey
    */
-  public void initRouter(MapzenRouter router, String apiKey) {
-    requestHandler.turnByTurnHandler().setApiKey(apiKey);
+  public void initRouter(MapzenRouter router) {
```

Comment:
do you think it make sense to provide an overloaded method to provide a different apikey than the one set for `MapzenMap`?

Reply:
I think that since we moved to the unified API key, providing different keys for each Mapzen service shouldn't be necessary. I would say we wait to do this until a specific use case or request comes in for it.